scripts/install

```diff
@@ -1,167 +1,354 @@
+### Have a look at the app to be sure this app needs a systemd script.
+### `ynh_systemd_config` will use the file conf/systemd.service
+### If you're not using these lines:
+###		- You can remove those files in conf/.
+###		- Remove the section "BACKUP SYSTEMD" in the backup script
+###		- Remove also the section "STOP AND REMOVE SERVICE" in the remove script
+###		- As well as the section "RESTORE SYSTEMD" in the restore script
+###		- And the section "SETUP SYSTEMD" in the upgrade script
+
+# Create a dedicated systemd config
+#ynh_add_systemd_config
+
+#=================================================
+# SETUP APPLICATION WITH CURL
+#=================================================
+
+### Use these lines only if the app installation needs to be finalized through
+### web forms. We generally don't want to ask the final user,
+### so we're going to use curl to automatically fill the fields and submit the
+### forms.
+
+# Set right permissions for curl install
+#chown -R $app: $final_path
+
+# Set the app as temporarily public for curl call
+#ynh_print_info "Configuring SSOwat..."
+#ynh_app_setting_set $app skipped_uris "/"
+# Reload SSOwat config
+#yunohost app ssowatconf
+
+# Reload Nginx
+#systemctl reload nginx
+
+# Installation with curl
+#ynh_print_info "Finalizing installation..."
+#ynh_local_curl "/INSTALL_PATH" "key1=value1" "key2=value2" "key3=value3"
+
+# Remove the public access
+#if [ $is_public -eq 0 ]
+#then
+#	ynh_app_setting_delete $app skipped_uris
+#fi
+
+#=================================================
+# MODIFY A CONFIG FILE
+#=================================================
+
+### `ynh_replace_string` is used to replace a string in a file.
+### (It's compatible with sed regular expressions syntax)
+
+#ynh_replace_string "match_string" "replace_string" "$final_path/CONFIG_FILE"
+
+#=================================================
+# STORE THE CONFIG FILE CHECKSUM
+#=================================================
+
+### `ynh_store_file_checksum` is used to store the checksum of a file.
+### That way, during the upgrade script, by using `ynh_backup_if_checksum_is_different`,
+### you can make a backup of this file before modifying it again if the admin had modified it.
+
+# Calculate and store the config file checksum into the app settings
+#ynh_store_file_checksum "$final_path/CONFIG_FILE"
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
 #=================================================
 
-sudo chown www-data:www-data $final_path -R
+### For security reason, any app should set the permissions to root: before anything else.
+### Then, if write authorization is needed, any access should be given only to directories
+### that really need such authorization.
+
+# Set permissions to app files
+sudo chown -R www-data: $final_path
 sudo chmod 777 $final_path -R
```

Comment:
Huh no, not a 777 !
Permission should be as most restrictive as possible.

Reply:
authorization removed